lib/Base/Organization/CollectionSorters.vala

```diff
@@ -0,0 +1,98 @@
+/* CollectionSorters.vala
+ *
+ * Copyright 2023 Frederick Schenk
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+using GLib;
+
+/**
+ * Interface for methods used to sort collections.
+ */
+public interface Backend.CollectionSorter <T> : Object {
+
+  /**
+   * Compares two items while sorting the list.
+   *
+   * @param a The first item to compare.
+   * @param b The second item to compare.
+   *
+   * @return How the items are sorted (positive when a before b, negative when b before a).
+   */
+  public abstract int compare_items (T a, T b);
+
+}
+
+/**
+ * Sorts a collection with posts reverse chronological.
+ *
+ * The sorting is mostly reverse chronological, with a few exceptions.
+ * Any item that is not a Post, particular PseudoItems, will be placed on top.
+ */
+public class Backend.ReversePostSorter : CollectionSorter <Object>, Object {
+
+  /**
+   * Compares two items while sorting the list.
+   *
+   * @param a The first item to compare.
+   * @param b The second item to compare.
+   *
+   * @return How the items are sorted (positive when a before b, negative when b before a).
+   */
+  protected int compare_items (Object a, Object b) {
+    // Attempt to cast as Posts
+    var post_a = a as Post;
+    var post_b = b as Post;
+    if (a == null || b == null) {
+      return compare_non_posts (a, b);
+    }
+
+    // Sort posts by date
+    DateTime x = post_a.creation_date;
+    DateTime y = post_b.creation_date;
+    return -1 * x.compare (y);
```

Comment:
Presumably `y.compare(x)` would avoid the need for negating it. Or do you find the `-1` clearer for showing that we're forcing a specific direction?

Reply:
I use [`GLib.DateTime.compare`](https://valadoc.org/glib-2.0/GLib.DateTime.compare.html) here, which would sort old to new. So, we do need to negate this.

Reply:
But if you flip the variables then that should (in most implementations) give you the opposite, so `y.compare(x)` wouldn't need multiplying 🙂 But it is easier to overlook the order, and `x.compare(y)` feels more "natural" because of alphabetical ordering.

Reply:
> But if you flip the variables then that should (in most implementations) give you the opposite, so y.compare(x) wouldn't need multiplying.

Seems like with `GLib.DateTime` it does the same. But then, I would still keep it that way as an indicator for the reverse direction.